ubuntu-advantage

```diff
@@ -141,6 +154,116 @@ disable_esm() {
     fi
 }
 
+fips_prep_check() {
+    local fips_hmacs pkg result
+
+    result=0
+    # sanity check
+    case "$ARCH" in
+        s390x|x86_64|ppc64le)
+            ;;
+        *)
+            echo "ERROR: $ARCH is not fips supported."
+            return 1
+            ;;
+    esac
+
+    # check to see if fips pkgs already installed.
+    fips_hmacs="openssh-client-hmac openssh-server-hmac libssl1.0.0-hmac \
+                linux-fips strongswan-hmac"
+    for pkg in $fips_hmacs; do
+        dpkg-query -s "$pkg" > /dev/null 2>&1 || result=$?
+        if [ $result -eq 0 ]; then
+            echo "ERROR: Looks like FIPS already installed."
+            return 1
+        fi
+    done
+
+    return 0
+}
+
+configure_fips() {
+    local bootdev fips_params result
+
+    result=0
+    # if /boot has its own partition, then get the bootdevice
+    # Note: /boot/efi  does not count
+    bootdev=$(awk '!/^\s*#/ && $2 ~ /^\/boot\/?$/ { print $1 }' $FSTAB)
+    fips_params="fips=1"
+    if [ -n "$bootdev" ]; then
+        fips_params="$fips_params bootdev=$bootdev"
+    fi
+
+    if [ "$ARCH" = "s390x" ]; then
+        sed -i -e 's/^parameters\s*=.*/& '"$fips_params"'/' "$FIPS_BOOT_CFG"
```

Comment:
I think you missed my previous comment about this:

You should not use `/` as sed s command separator since `$bootdev` can also contain `/`.
I usually use `s,<re>,<sub>,` in these cases.

Reply:
Oh gosh, my apologies, I did miss that one. Let me fix it now. And I also need to remove the brackets from README since fips does not yet have a website.